src/tests/unit/sync/syncimages_status.f90

```diff
@@ -10,7 +10,6 @@ program sync_images_stat
   me = this_image()
 
   if (me /= 1 ) then
-     call sleep(1)
```

Comment:
Nearly correct. Only the reason for the race is not absolutely correct, but a little bit more difficult to explain. In the former implementation the status of the image was read using one-sided communication. When that read was before the aborting image set its abort-status, then the program would hang. When the set of the abort-status was interleaved with some images having read the status already and some not, then the race begun, that the images that had the abort received set their own abort-status while the other images where reading that status. So timing was everything here.

Now the abort-status is send using two-sided communication, which also means, that the sync is done at this stage. That way no race is possible anymore, because an aborted-image would publish its state and all other receive it a defined location.